### PR TITLE
Review fixes

### PR DESCRIPTION
**Pull request by kpp**

### CODE REVIEW

File: transports/quic/tests/smoke.rs [hunk 1]
```diff
@@ -27,13 +36,33 @@ async fn smoke_tls() -> Result<()> {
...
+        generate_noise_keypair()
+    }
+}
+
+#[cfg(feature = "tls")]
+impl GenerateKeypair for libp2p_quic::TlsCrypto {
+    fn generate_keypair() -> Self::Keypair {
+        generate_tls_keypair()
+    }
+}
```
Comment (dvc94ch) [inline transports/quic/tests/smoke.rs:55]: would it make sense to move this to the crypto trait? seems useful, and especially for ed25519_dalek, it's really annoying that you have to match the rand version. this makes sure that applications don't have to depend on a specific rand.

**Review by dvc94ch — APPROVED** [review]
Looks good! thanks

**Comment by dvc94ch** [comment]
Also there are still to missing things:
- No tests for /ipV/ADDR/udp/PORt/quic/p2p/PEERID.
- Criticality can be enabled again according to libp2p/go-libp2p-tls#87

File: transports/quic/src/transport.rs [hunk 2]
```diff
@@ -292,17 +289,23 @@ pub(crate) fn socketaddr_to_multiaddr(socket_addr: &SocketAddr) -> Multiaddr {
 mod tests {
     use super::*;
 
+    #[cfg(feature = "noise")]
+    type Crypto = crate::NoiseCrypto;
+    #[cfg(feature = "tls")]
+    type Crypto = crate::TlsCrypto;
```
Comment (dvc94ch) [inline transports/quic/src/transport.rs:295]: this won't work when both features are enabled

File: transports/quic/tests/smoke.rs (hunk 1 above)
Comment (kpp) [reply transports/quic/tests/smoke.rs:55]: I don’t think so. There is no need

File: transports/quic/src/transport.rs (hunk 2 above)
Comment (kpp) [reply transports/quic/src/transport.rs]: Done

**Pull request merged**